velox/exec/MergeJoin.cpp

```diff
@@ -164,6 +168,11 @@ bool MergeJoin::needsInput() const {
 }
 
 void MergeJoin::addInput(RowVectorPtr input) {
+  // TODO: Ensuring everything gets materialized upfront for now. If this
```

Comment:
Is this a change from before? For highly-selective joins, can we avoid loading non-key columns or columns that are used only in the filter?

Reply:
Good point. I can materialize them somewhere else to prevent this case. I remember this was a limitation somewhere else I started hitting when lazy vectors were propagated, but my memory is failing me.

I'll try it again and update.